fdbserver/ClusterController.actor.cpp

```diff
@@ -248,6 +248,7 @@ ACTOR Future<Void> clusterWatchDatabase(ClusterControllerData* cluster,
 			dbInfo.latencyBandConfig = db->serverInfo->get().latencyBandConfig;
 			dbInfo.myLocality = db->serverInfo->get().myLocality;
 			dbInfo.client = ClientDBInfo();
+			dbInfo.client.isEncryptionEnabled = SERVER_KNOBS->ENABLE_ENCRYPTION;
```

Comment:
@sfc-gh-ajbeamon For encryption "Tenant" or its support by the cluster is prerequisite, do we need to add an extra check along with `SERVER_KNOB->ENABLE_ENCRYPTION` (here and elsewhere) to make this invariant hold all the time?

Reply:
I don't exactly understand the question. Can you clarify?

Reply:
For Encryption cluster `tenant` support is a pre-requisite, I just noticed that the KNOB doesn't account for the `tenant support` being enabled (Or is there such a concept) which feel incorrect. Any thoughts how to incorporate `enable_encryption` knob to depend on `cluster tenant support` property?

Reply:
The cluster controller tracks the current tenant state of the cluster, which is part of the `ClientDBInfo` as `tenantMode`. If we only want to enable encryption if the `tenantMode` is `required`, I think we could do that.

Reply:
Is it possible to have dependency managed at the time of initialization? For instance:

`init( ENABLE_TLOG_ENCRYPTION,                  ENABLE_ENCRYPTION );`

Note: Encryption knob is defined in SERVER_KNOB.

Reply:
Btw, we can take this as a separate PR. I'm happy to `stamp` this PR, unless, AJ has more suggestions, please merge it when possible.

Reply:
You mean at the time of knob initialization? I don't think so, the configuration comes from the transaction state store and can be changed while the cluster is running.

Reply:
Makes sense. 
@sfc-gh-nwijetunga it would be ideal to think of a way to expose a routine `isEncryptionEnabled` and use it across the board marrying two knobs together to give a consistent (and correct) encryption operational mode (as a separate PR offcourse). Thoughts?

Thanks

Reply:
Yeah I can work on that next